server/backend/api/app/routers/router_guards.py

```diff
@@ -262,59 +225,86 @@ async def authorize_by_privilege(
     """
-    # from path_param - user_role_xref_id
+    # from path_param - "user_role_xref_id"; should exists already in db.
     if "user_role_xref_id" in request.path_params:
-        user_role = crud_user_role.find_by_id(
-            db, request.path_params["user_role_xref_id"]
-        )
-        return (
-            TargetUser.model_validate(user_role.user) if user_role is not None else None
-        )
+        urxid = request.path_params["user_role_xref_id"]
+        LOGGER.debug("Dependency 'get_target_user_from_id' called with " +
+                     f"request containing user_role_xref_id path param {urxid}.")
+        user_role = crud_user_role.find_by_id(db, urxid)
+        if user_role is not None:
+            found_target_user = TargetUser.model_validate(user_role.user)
+            return found_target_user
+        else:
+            error_msg = "Parameter 'user_role_xref_id' is missing or invalid."
+            raise_http_exception(
+                error_code=ERROR_CODE_INVALID_REQUEST_PARAMETER,
+                error_msg=error_msg
+            )
+
     else:
         # from request body - {user_name/user_type_code}
         rbody = await request.json()
+        LOGGER.debug("Dependency 'get_target_user_from_id' called with " +
+                     f"request body {rbody}.")
+        rb_user_name = rbody["user_name"]
+        rb_user_type_code = rbody["user_type_code"]
+        # user_guid is searched from IDIM search and passed from frontend.
+        rb_user_guid = rbody["user_guid"]
+
         user = crud_user.get_user_by_domain_and_name(
             db,
-            rbody["user_type_code"],
-            rbody["user_name"],
+            rb_user_type_code,
+            rb_user_name,
         )
-        return TargetUser.model_validate(user) if user is not None else None
+        if user is not None:
+            found_target_user = TargetUser.model_validate(user)
+            # old data might not have user_guid.
+            if found_target_user.user_guid is None:
+                found_target_user.user_guid = rb_user_guid
+            return found_target_user
+
+        # user not found, case of new user.
+        else:
+            target_new_user = TargetUser.model_validate({
+                "user_name": rb_user_name,
+                "user_type_code": rb_user_type_code,
+                "user_guid": rb_user_guid
+            })
+            return target_new_user
 
 
 async def authorize_by_user_type(
     request: Request,
     requester: Requester = Depends(get_current_requester),
-    target_user: Union[TargetUser, None] = Depends(get_target_user_from_id),
+    target_user: TargetUser = Depends(get_target_user_from_id),
 ):
     """
     This authorize_by_user_type method is used to forbidden business bceid user manage idir user's access
     """
     if requester.user_type_code == UserType.BCEID:
         target_user_type_code = None
-        if target_user:
+        if not target_user.is_new_user():
```

Comment:
Hi Ian, I think here we can just get `target_user.user_type_code`, no need the if-else condition, it should have it, otherwise we throw error below

Reply:
That's right, this is adjusted.